doc/releases/release-notes-3.6.rst

```diff
@@ -363,6 +383,10 @@ Drivers and Sensors
   * Fixed sam_gmac driver. PTP clock adjustment was wrong for negative values.
   * Fixed adin2111 driver. Initialization was done incorrectly when working with adin2110.
   * Fixed ksz8081 driver. Logging changes, RMII clock fixes, GPIO pin fixes.
+  * Added a driver for NXP ENET which is a rework of the old ``eth_mcux.c`` driver which
+    had become unmaintainable due to fundamental problems with the lack of PHY abstraction.
+    ``eth_mcux.c`` and the corresponding compatible ``nxp,kinetis-ethernet`` will be deprecated
+    and this new driver will be supported instead.
```

Comment:
Probably deserves a mention in the migration guide

Reply:
@decsny will work to add it to the migration guide.

Reply:
@mmahadevan108, this line item needs "EXPERIMENTAL" as this driver is still in experimental state.

@MaureenHelm, if the driver is experimental, would we put something in the migration guide to highlight how to use it?

Reply:
The migration guide is only for instructions needed for migrating. I'd out experimental drivers in the release notes.

Reply:
> I'd out experimental drivers in the release notes.

Not sure what you are saying...

Reply:
Sorry, meant to write "put", not "out".

Reply:
I have added a mention that this driver is experimental.